target/models.py

```diff
@@ -20,3 +23,16 @@ class Target(models.Model):
 
     def __str__(self):
         return f'{self.pk} - {self.user.email}, {self.title}'
+
+    def get_matches(self):
+        return Target.objects.filter(
+            location__distance_lte=(self.location, self.radius + F('radius')),
+            topic=self.topic,
+        ).exclude(user=self.user)
+
+
+@receiver(post_save, sender=Target)
+def chat_manager(sender, **kwargs):
+    from target.target_utils import manage_target_chats
+    target = kwargs.get('instance')
+    manage_target_chats(target)
```

Comment:
I'm not fan of signal but it allows me to do exactly what y wanted. I must run the function `manage_target_chats ` when a target is saved (created or edited). I just added this code instead of duplicating those lines in several places. 
In order to make it visible I place it below its model, so when you see the model you will notice that some code runs after saving.
As the code is in another file, just for clean code reasons, I have to import 'manage_target_chats' locally in order to prevent recursive import problems.

Reply:
I agree that in this case maybe it's the better option, and since it's on the same model, it should be ok to debug if something goes wrong.
Are you aware that if somewhere in the code you do `Target.objects.update(...)` this method will not trigger then.

Reply:
Good catch, you are right, [Django doc](https://docs.djangoproject.com/en/3.0/topics/db/queries/#updating-multiple-objects-at-once) explains perfectly why the signal is not called. I think there is no problem in this case as we don't need to do an explicit `update`. 
But anyway, how we could manage this? I mean, if I override the save method would happen the same, the save method will be not called when doing an update.